File: backend/schemas/objetivo.py
"""
Schemas Pydantic para el módulo de objetivos.
ObjetivoCreate → Update → Response → ListResponse

responsable_id: FK a users (operadores RRHH), NO empleados.
empresa_id: explícito en Create; heredado por el objeto en lecturas.
estado: no se pide al crear (default 'por_hacer'); se cambia por CambiarEstadoRequest.
"""
from datetime import date, datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel

PRIORIDADES = {"baja", "media", "alta"}
ESTADOS = {"por_hacer", "haciendo", "terminado"}
ORDEN_ESTADOS = ["por_hacer", "haciendo", "terminado"]


class ObjetivoCreate(BaseModel):
    empresa_id:     UUID
    responsable_id: UUID
    titulo:         str
    descripcion:    Optional[str] = None
    prioridad:      str = "media"
    fecha_entrega:  Optional[date] = None


class ObjetivoUpdate(BaseModel):
    responsable_id: Optional[UUID] = None
    titulo:         Optional[str] = None
    descripcion:    Optional[str] = None
    prioridad:      Optional[str] = None
    fecha_entrega:  Optional[date] = None


class CambiarEstadoRequest(BaseModel):
    estado: str


class ObjetivoResponse(BaseModel):
    id:                  str
    empresa_id:          str
    empresa_nombre:      Optional[str] = None
    responsable_id:      str
    responsable_nombre:  Optional[str] = None
    titulo:              str
    descripcion:         Optional[str] = None
    prioridad:           str
    estado:              str
    fecha_entrega:       Optional[date] = None
    created_at:          datetime
    updated_at:          datetime


class ObjetivoListResponse(BaseModel):
    items: List[ObjetivoResponse]
    total: int

File: frontend/types/objetivo.ts
export type EstadoObjetivo  = "por_hacer" | "haciendo" | "terminado"
export type PrioridadObjetivo = "baja" | "media" | "alta"

export interface Objetivo {
  id: string
  empresa_id: string
  empresa_nombre: string | null
  responsable_id: string
  responsable_nombre: string | null
  titulo: string
  descripcion: string | null
  prioridad: PrioridadObjetivo
  estado: EstadoObjetivo
  fecha_entrega: string | null  // "YYYY-MM-DD"
  created_at: string
  updated_at: string
}

export interface ObjetivoCreate {
  empresa_id: string
  responsable_id: string
  titulo: string
  descripcion?: string
  prioridad: PrioridadObjetivo
  fecha_entrega?: string
}

export interface ObjetivoUpdate {
  responsable_id?: string
  titulo?: string
  descripcion?: string
  prioridad?: PrioridadObjetivo
  fecha_entrega?: string
}

export interface CambiarEstadoRequest {
  estado: EstadoObjetivo
}

export interface ObjetivoListResponse {
  items: Objetivo[]
  total: number
}

export interface UserItem {
  id: string
  nombre: string
  apellido: string
  email: string
  rol: string
}
